### ISSUE TITLE
one of three didnt workm [Setup of platform gree is taking longer than 60 seconds. Startup will proceed without waiting any longer]

### ISSUE THREAD

**Issue by MukhaBoyka** [RobHofmann/HomeAssistant-GreeClimateComponent#164]
**Describe the bug**

I have three Air Conditioners, two of them(cooper & hunter **CH-S09FTXE (WIFI)** )  i boght 2 years ago and add to HA with this integration.
Now, month ago bought new one (cooper & hunter **CH-S07FTXQ2-NG (WIFI)**) and have a problem with connecting it.

**Configuration**
```
- platform: gree # this one have a problem! 
  name: Kitchen AC WIFI
  host: 10.0.0.28
  port: 7000
  mac: #MAC#
  target_temp_step: 1
  #uid: 10001
  temp_sensor: sensor.kitchen_ac_wifi_temperature
  lights: input_boolean.kitchen_ac_wifi_lights
  xfan: input_boolean.kitchen_ac_wifi_xfan
  health: input_boolean.kitchen_ac_wifi_health
  sleep: input_boolean.kitchen_ac_wifi_sleep
  powersave: input_boolean.kitchen_ac_wifi_powersave
  eightdegheat: input_boolean.kitchen_ac_wifi_8degree
  air: input_boolean.kitchen_ac_wifi_air_scavenging

- platform: gree
  name: Bedroom AC WIFI
  host: 10.0.0.251
  port: 7000
  mac: #MAC#
  target_temp_step: 1
  #uid: 1000251
  temp_sensor: sensor.bedroom_ac_wifi_temperature
  lights: input_boolean.bedroom_ac_wifi_lights
  xfan: input_boolean.bedroom_ac_wifi_xfan
  health: input_boolean.bedroom_ac_wifi_health
  sleep: input_boolean.bedroom_ac_wifi_sleep
  powersave: input_boolean.bedroom_ac_wifi_powersave
  eightdegheat: input_boolean.bedroom_ac_wifi_8degree
  air: input_boolean.bedroom_ac_wifi_air_scavenging

- platform: gree
  name: Small Room AC WIFI
  host: 10.0.0.96
  port: 7000
  mac: #MAC#
  target_temp_step: 1
  #uid: 100096
  temp_sensor: sensor.small_room_ac_wifi_temperature
  lights: input_boolean.small_room_ac_wifi_lights
  xfan: input_boolean.small_room_ac_wifi_xfan
  health: input_boolean.small_room_ac_wifi_health
  sleep: input_boolean.small_room_ac_wifi_sleep
  powersave: input_boolean.small_room_ac_wifi_powersave
  eightdegheat: input_boolean.small_room_ac_wifi_8degree
  air: input_boolean.small_room_ac_wifi_air_scavenging
```

**Screenshots**

**Platform:**
- Raspberry Pi 4
- Core 2024.5.5
- Supervisor 2024.05.1
- Operating System 12.3
- Frontend 20240501.1

**Additional context**

**Logs**
```
2024-05-25 16:51:46.035 INFO (MainThread) [custom_components.gree.climate] Setting up Gree climate platform
2024-05-25 16:51:46.035 INFO (MainThread) [custom_components.gree.climate] Adding Gree climate device to hass
2024-05-25 16:51:46.035 INFO (MainThread) [custom_components.gree.climate] Initialize the GREE climate device
2024-05-25 16:51:46.035 INFO (MainThread) [custom_components.gree.climate] Retrieving HVAC encryption key
2024-05-25 16:51:46.041 INFO (MainThread) [custom_components.gree.climate] Fetching(10.0.0.28, 7000, 10, {"cid": "app","i": 1,"pack": "c59unMPoEOqeuDiax+ZmW4ffbs8y/J7sG2kk5G8CCK+j9mrs/0p7GUTdWxmNaBUt","t":"pack","tcid":"9424b8b5a1cc","uid": 0})
2024-05-25 16:51:56.057 ERROR (MainThread) [homeassistant.components.climate] Setup of platform gree is taking longer than 60 seconds. Startup will proceed without waiting any longer.
2024-05-25 16:51:56.058 INFO (MainThread) [custom_components.gree.climate] Setting up Gree climate platform
2024-05-25 16:51:56.059 INFO (MainThread) [custom_components.gree.climate] Adding Gree climate device to hass
2024-05-25 16:51:56.059 INFO (MainThread) [custom_components.gree.climate] Initialize the GREE climate device
2024-05-25 16:51:56.059 INFO (MainThread) [custom_components.gree.climate] Retrieving HVAC encryption key
2024-05-25 16:51:56.060 INFO (MainThread) [custom_components.gree.climate] Fetching(10.0.0.251, 7000, 10, {"cid": "app","i": 1,"pack": "tyxS3K7bQkBIw5lXGdvLl3gOiAzttphuNu1XE/fVtm+j9mrs/0p7GUTdWxmNaBUt","t":"pack","tcid":"502cc63d1645","uid": 0})
2024-05-25 16:51:56.075 INFO (MainThread) [custom_components.gree.climate] Fetched device encrytion key: b'w9Yz2Bc5Ef8Hi1Kl'
2024-05-25 16:51:56.075 INFO (MainThread) [custom_components.gree.climate] Setting up temperature sensor: sensor.bedroom_ac_wifi_temperature
2024-05-25 16:51:56.076 INFO (MainThread) [custom_components.gree.climate] Setting up lights entity: input_boolean.bedroom_ac_wifi_lights
2024-05-25 16:51:56.076 INFO (MainThread) [custom_components.gree.climate] Setting up xfan entity: input_boolean.bedroom_ac_wifi_xfan
2024-05-25 16:51:56.076 INFO (MainThread) [custom_components.gree.climate] Setting up health entity: input_boolean.bedroom_ac_wifi_health
2024-05-25 16:51:56.076 INFO (MainThread) [custom_components.gree.climate] Setting up powersave entity: input_boolean.bedroom_ac_wifi_powersave
2024-05-25 16:51:56.076 INFO (MainThread) [custom_components.gree.climate] Setting up sleep entity: input_boolean.bedroom_ac_wifi_sleep
2024-05-25 16:51:56.076 INFO (MainThread) [custom_components.gree.climate] Setting up 8℃ heat entity: input_boolean.bedroom_ac_wifi_8degree
2024-05-25 16:51:56.076 INFO (MainThread) [custom_components.gree.climate] Setting up air entity: input_boolean.bedroom_ac_wifi_air_scavenging
2024-05-25 16:51:56.077 INFO (MainThread) [custom_components.gree.climate] name(): Bedroom AC WIFI
2024-05-25 16:51:56.077 INFO (MainThread) [custom_components.gree.climate] name(): Bedroom AC WIFI
2024-05-25 16:51:56.077 INFO (MainThread) [custom_components.gree.climate] supported_features(): 441
2024-05-25 16:51:56.077 INFO (MainThread) [custom_components.gree.climate] temperature_unit(): °C
2024-05-25 16:51:56.078 INFO (MainThread) [custom_components.gree.climate] hvac_modes(): [<HVACMode.AUTO: 'auto'>, <HVACMode.COOL: 'cool'>, <HVACMode.DRY: 'dry'>, <HVACMode.FAN_ONLY: 'fan_only'>, <HVACMode.HEAT: 'heat'>, <HVACMode.OFF: 'off'>]
2024-05-25 16:51:56.078 INFO (MainThread) [custom_components.gree.climate] min_temp(): 16
2024-05-25 16:51:56.078 INFO (MainThread) [custom_components.gree.climate] max_temp(): 30
2024-05-25 16:51:56.078 INFO (MainThread) [custom_components.gree.climate] target_temperature_step(): 1.0
2024-05-25 16:51:56.078 INFO (MainThread) [custom_components.gree.climate] fan_list(): ['Auto', 'Low', 'Medium-Low', 'Medium', 'Medium-High', 'High', 'Turbo', 'Quiet']
2024-05-25 16:51:56.078 INFO (MainThread) [custom_components.gree.climate] preset_modes(): ['Default', 'Full swing', 'Fixed in the leftmost position', 'Fixed in the middle-left position', 'Fixed in the middle postion', 'Fixed in the middle-right position', 'Fixed in the rightmost position']
2024-05-25 16:51:56.078 INFO (MainThread) [custom_components.gree.climate] swing_modes(): ['Default', 'Swing in full range', 'Fixed in the upmost position', 'Fixed in the middle-up position', 'Fixed in the middle position', 'Fixed in the middle-low position', 'Fixed in the lowest position', 'Swing in the downmost region', 'Swing in the middle-low region', 'Swing in the middle region', 'Swing in the middle-up region', 'Swing in the upmost region']
2024-05-25 16:51:56.079 INFO (MainThread) [custom_components.gree.climate] supported_features(): 441
2024-05-25 16:51:56.079 INFO (MainThread) [custom_components.gree.climate] Gree climate device added to hass()
2024-05-25 16:51:56.079 INFO (MainThread) [custom_components.gree.climate] Starting SyncState
2024-05-25 16:51:56.080 INFO (MainThread) [custom_components.gree.climate] Fetching(10.0.0.251, 7000, 10, {"cid":"app","i":0,"pack":"NvxmDQ8YDIC0JgjCkLAyhJfmKhHTV/Cpu1wR0Gnj4YU979ZsPLeQDhX2jeNpaol2pQobL9wiWhYhJzQrEE4DkztF/DH3Jowqvdb+qy6FDYvEX4CyRmS01Vui8TanlqF8uOg8pmijYWTbf0KXHFMgquGSO75ZttzxdYZAZQmTWoY8bdonUutd+BoU07Uql9DBrpUf8Hkrn9OWhD2vZ2j6DzuYHSWMqj48fPByhulvT7vA4CfzroFPCFip3PsdbTdYUYCS/xjhN9cSZKWkxTQdiVJXFBIf76FcAXagWJHSPNA=","t":"pack","tcid":"502cc63d1645","uid":0})
2024-05-25 16:51:56.104 INFO (MainThread) [custom_components.gree.climate] Setting acOptions with retrieved HVAC values
2024-05-25 16:51:56.104 INFO (MainThread) [custom_components.gree.climate] Setting Pow: 1
2024-05-25 16:51:56.104 INFO (MainThread) [custom_components.gree.climate] Setting Mod: 0
2024-05-25 16:51:56.105 INFO (MainThread) [custom_components.gree.climate] Setting SetTem: 25
2024-05-25 16:51:56.105 INFO (MainThread) [custom_components.gree.climate] Setting WdSpd: 0
2024-05-25 16:51:56.105 INFO (MainThread) [custom_components.gree.climate] Setting Air: 0
2024-05-25 16:51:56.105 INFO (MainThread) [custom_components.gree.climate] Setting Blo: 0
2024-05-25 16:51:56.105 INFO (MainThread) [custom_components.gree.climate] Setting Health: 0
2024-05-25 16:51:56.105 INFO (MainThread) [custom_components.gree.climate] Setting SwhSlp: 0
2024-05-25 16:51:56.105 INFO (MainThread) [custom_components.gree.climate] Setting Lig: 1
2024-05-25 16:51:56.105 INFO (MainThread) [custom_components.gree.climate] Setting SwingLfRig: 0
2024-05-25 16:51:56.105 INFO (MainThread) [custom_components.gree.climate] Setting SwUpDn: 2
2024-05-25 16:51:56.105 INFO (MainThread) [custom_components.gree.climate] Setting Quiet: 0
2024-05-25 16:51:56.105 INFO (MainThread) [custom_components.gree.climate] Setting Tur: 0
2024-05-25 16:51:56.105 INFO (MainThread) [custom_components.gree.climate] Setting StHt: 0
2024-05-25 16:51:56.105 INFO (MainThread) [custom_components.gree.climate] Setting TemUn: 0
2024-05-25 16:51:56.105 INFO (MainThread) [custom_components.gree.climate] Setting HeatCoolType: 0
2024-05-25 16:51:56.105 INFO (MainThread) [custom_components.gree.climate] Setting TemRec: 0
2024-05-25 16:51:56.106 INFO (MainThread) [custom_components.gree.climate] Setting SvSt: 0
2024-05-25 16:51:56.106 INFO (MainThread) [custom_components.gree.climate] Setting SlpMod: 0
2024-05-25 16:51:56.106 INFO (MainThread) [custom_components.gree.climate] Done setting acOptions
2024-05-25 16:51:56.106 INFO (MainThread) [custom_components.gree.climate] HA target temp set according to HVAC state to: 25
2024-05-25 16:51:56.106 INFO (MainThread) [custom_components.gree.climate] HA lights option set according to HVAC state to: on
2024-05-25 16:51:56.106 INFO (MainThread) [custom_components.gree.climate] HA xfan option set according to HVAC state to: off
2024-05-25 16:51:56.106 INFO (MainThread) [custom_components.gree.climate] HA health option set according to HVAC state to: off
2024-05-25 16:51:56.106 INFO (MainThread) [custom_components.gree.climate] HA powersave option set according to HVAC state to: off
2024-05-25 16:51:56.106 INFO (MainThread) [custom_components.gree.climate] HA sleep option set according to HVAC state to: off
2024-05-25 16:51:56.107 INFO (MainThread) [custom_components.gree.climate] HA 8℃ heat option set according to HVAC state to: off
2024-05-25 16:51:56.107 INFO (MainThread) [custom_components.gree.climate] HA air option set according to HVAC state to: off
2024-05-25 16:51:56.107 INFO (MainThread) [custom_components.gree.climate] HA operation mode set according to HVAC state to: auto
2024-05-25 16:51:56.107 INFO (MainThread) [custom_components.gree.climate] HA swing mode set according to HVAC state to: Fixed in the upmost position
2024-05-25 16:51:56.107 INFO (MainThread) [custom_components.gree.climate] HA preset mode set according to HVAC state to: Default
2024-05-25 16:51:56.108 INFO (MainThread) [custom_components.gree.climate] HA fan mode set according to HVAC state to: Auto
2024-05-25 16:51:56.108 INFO (MainThread) [custom_components.gree.climate] Finished SyncState
2024-05-25 16:51:56.108 INFO (MainThread) [custom_components.gree.climate] supported_features(): 441
2024-05-25 16:51:56.108 INFO (MainThread) [custom_components.gree.climate] temperature_unit(): °C
2024-05-25 16:51:56.108 INFO (MainThread) [custom_components.gree.climate] hvac_modes(): [<HVACMode.AUTO: 'auto'>, <HVACMode.COOL: 'cool'>, <HVACMode.DRY: 'dry'>, <HVACMode.FAN_ONLY: 'fan_only'>, <HVACMode.HEAT: 'heat'>, <HVACMode.OFF: 'off'>]
2024-05-25 16:51:56.109 INFO (MainThread) [custom_components.gree.climate] min_temp(): 16
2024-05-25 16:51:56.109 INFO (MainThread) [custom_components.gree.climate] max_temp(): 30
2024-05-25 16:51:56.109 INFO (MainThread) [custom_components.gree.climate] target_temperature_step(): 1.0
2024-05-25 16:51:56.109 INFO (MainThread) [custom_components.gree.climate] fan_list(): ['Auto', 'Low', 'Medium-Low', 'Medium', 'Medium-High', 'High', 'Turbo', 'Quiet']
2024-05-25 16:51:56.109 INFO (MainThread) [custom_components.gree.climate] preset_modes(): ['Default', 'Full swing', 'Fixed in the leftmost position', 'Fixed in the middle-left position', 'Fixed in the middle postion', 'Fixed in the middle-right position', 'Fixed in the rightmost position']
2024-05-25 16:51:56.110 INFO (MainThread) [custom_components.gree.climate] swing_modes(): ['Default', 'Swing in full range', 'Fixed in the upmost position', 'Fixed in the middle-up position', 'Fixed in the middle position', 'Fixed in the middle-low position', 'Fixed in the lowest position', 'Swing in the downmost region', 'Swing in the middle-low region', 'Swing in the middle region', 'Swing in the middle-up region', 'Swing in the upmost region']
2024-05-25 16:51:56.110 INFO (MainThread) [custom_components.gree.climate] hvac_mode(): auto
2024-05-25 16:51:56.111 INFO (MainThread) [custom_components.gree.climate] supported_features(): 441
2024-05-25 16:51:56.111 INFO (MainThread) [custom_components.gree.climate] temperature_unit(): °C
2024-05-25 16:51:56.111 INFO (MainThread) [custom_components.gree.climate] current_temperature(): None
2024-05-25 16:51:56.111 INFO (MainThread) [custom_components.gree.climate] target_temperature(): 25
2024-05-25 16:51:56.112 INFO (MainThread) [custom_components.gree.climate] fan_mode(): Auto
2024-05-25 16:51:56.112 INFO (MainThread) [custom_components.gree.climate] preset_mode(): Default
2024-05-25 16:51:56.112 INFO (MainThread) [custom_components.gree.climate] swing_mode(): Fixed in the upmost position
2024-05-25 16:51:56.112 INFO (MainThread) [custom_components.gree.climate] name(): Bedroom AC WIFI
2024-05-25 16:51:56.112 INFO (MainThread) [custom_components.gree.climate] supported_features(): 441
2024-05-25 16:51:56.113 INFO (MainThread) [custom_components.gree.climate] should_poll()
2024-05-25 16:51:56.114 INFO (MainThread) [custom_components.gree.climate] Setting up Gree climate platform
2024-05-25 16:51:56.115 INFO (MainThread) [custom_components.gree.climate] Adding Gree climate device to hass
2024-05-25 16:51:56.115 INFO (MainThread) [custom_components.gree.climate] Initialize the GREE climate device
2024-05-25 16:51:56.115 INFO (MainThread) [custom_components.gree.climate] Retrieving HVAC encryption key
2024-05-25 16:51:56.116 INFO (MainThread) [custom_components.gree.climate] Fetching(10.0.0.96, 7000, 10, {"cid": "app","i": 1,"pack": "KC6x+z09KxOhbN2srcj5uJbqkEtGIHAwAWe/5YRGxrmj9mrs/0p7GUTdWxmNaBUt","t":"pack","tcid":"f4911e47932b","uid": 0})
2024-05-25 16:51:56.134 INFO (MainThread) [custom_components.gree.climate] Fetched device encrytion key: b'Wx8Za1Cd4Fg7Ij0L'
2024-05-25 16:51:56.134 INFO (MainThread) [custom_components.gree.climate] Setting up temperature sensor: sensor.small_room_ac_wifi_temperature
2024-05-25 16:51:56.134 INFO (MainThread) [custom_components.gree.climate] Setting up lights entity: input_boolean.small_room_ac_wifi_lights
2024-05-25 16:51:56.134 INFO (MainThread) [custom_components.gree.climate] Setting up xfan entity: input_boolean.small_room_ac_wifi_xfan
2024-05-25 16:51:56.134 INFO (MainThread) [custom_components.gree.climate] Setting up health entity: input_boolean.small_room_ac_wifi_health
2024-05-25 16:51:56.134 INFO (MainThread) [custom_components.gree.climate] Setting up powersave entity: input_boolean.small_room_ac_wifi_powersave
2024-05-25 16:51:56.135 INFO (MainThread) [custom_components.gree.climate] Setting up sleep entity: input_boolean.small_room_ac_wifi_sleep
2024-05-25 16:51:56.135 INFO (MainThread) [custom_components.gree.climate] Setting up 8℃ heat entity: input_boolean.small_room_ac_wifi_8degree
2024-05-25 16:51:56.135 INFO (MainThread) [custom_components.gree.climate] Setting up air entity: input_boolean.small_room_ac_wifi_air_scavenging
2024-05-25 16:51:56.135 INFO (MainThread) [custom_components.gree.climate] name(): Small Room AC WIFI
2024-05-25 16:51:56.135 INFO (MainThread) [custom_components.gree.climate] name(): Small Room AC WIFI
2024-05-25 16:51:56.135 INFO (MainThread) [custom_components.gree.climate] supported_features(): 441
2024-05-25 16:51:56.136 INFO (MainThread) [custom_components.gree.climate] temperature_unit(): °C
2024-05-25 16:51:56.136 INFO (MainThread) [custom_components.gree.climate] hvac_modes(): [<HVACMode.AUTO: 'auto'>, <HVACMode.COOL: 'cool'>, <HVACMode.DRY: 'dry'>, <HVACMode.FAN_ONLY: 'fan_only'>, <HVACMode.HEAT: 'heat'>, <HVACMode.OFF: 'off'>]
2024-05-25 16:51:56.136 INFO (MainThread) [custom_components.gree.climate] min_temp(): 16
2024-05-25 16:51:56.136 INFO (MainThread) [custom_components.gree.climate] max_temp(): 30
2024-05-25 16:51:56.136 INFO (MainThread) [custom_components.gree.climate] target_temperature_step(): 1.0
2024-05-25 16:51:56.136 INFO (MainThread) [custom_components.gree.climate] fan_list(): ['Auto', 'Low', 'Medium-Low', 'Medium', 'Medium-High', 'High', 'Turbo', 'Quiet']
2024-05-25 16:51:56.136 INFO (MainThread) [custom_components.gree.climate] preset_modes(): ['Default', 'Full swing', 'Fixed in the leftmost position', 'Fixed in the middle-left position', 'Fixed in the middle postion', 'Fixed in the middle-right position', 'Fixed in the rightmost position']
2024-05-25 16:51:56.136 INFO (MainThread) [custom_components.gree.climate] swing_modes(): ['Default', 'Swing in full range', 'Fixed in the upmost position', 'Fixed in the middle-up position', 'Fixed in the middle position', 'Fixed in the middle-low position', 'Fixed in the lowest position', 'Swing in the downmost region', 'Swing in the middle-low region', 'Swing in the middle region', 'Swing in the middle-up region', 'Swing in the upmost region']
2024-05-25 16:51:56.136 INFO (MainThread) [custom_components.gree.climate] supported_features(): 441
2024-05-25 16:51:56.137 INFO (MainThread) [custom_components.gree.climate] Gree climate device added to hass()
2024-05-25 16:51:56.137 INFO (MainThread) [custom_components.gree.climate] Starting SyncState
2024-05-25 16:51:56.137 INFO (MainThread) [custom_components.gree.climate] Fetching(10.0.0.96, 7000, 10, {"cid":"app","i":0,"pack":"XdeKd6gUqocAO1FYikLhAfIZ5NnyU5jRYkybasPbk8d/UngW5+9L16VVbY5ByMI8fUf6+SELWCh/IeihDL/FVsCuyWrjYKTrgb8GMtHYQN67/i6X6y1iZog6CGkMEffraHcftaI/GhQR4KxxR/thpQaj5SfBAizaP4MBgNyhi+SmAUXZ6aIqxynaNdsybfBACxeZitHno2PpYnQMV28DA7ILjZd8JDiWIbcthRBVxU1yCzIwU7w+K0U6eOG91zmY1HUP/HVAVb76HSchJYJZVUPXeawCT3evR+SmE7vokig=","t":"pack","tcid":"f4911e47932b","uid":0})
2024-05-25 16:51:56.156 INFO (MainThread) [custom_components.gree.climate] Setting acOptions with retrieved HVAC values
2024-05-25 16:51:56.156 INFO (MainThread) [custom_components.gree.climate] Setting Pow: 1
2024-05-25 16:51:56.156 INFO (MainThread) [custom_components.gree.climate] Setting Mod: 0
2024-05-25 16:51:56.157 INFO (MainThread) [custom_components.gree.climate] Setting SetTem: 25
2024-05-25 16:51:56.157 INFO (MainThread) [custom_components.gree.climate] Setting WdSpd: 0
2024-05-25 16:51:56.157 INFO (MainThread) [custom_components.gree.climate] Setting Air: 0
2024-05-25 16:51:56.157 INFO (MainThread) [custom_components.gree.climate] Setting Blo: 0
2024-05-25 16:51:56.157 INFO (MainThread) [custom_components.gree.climate] Setting Health: 1
2024-05-25 16:51:56.157 INFO (MainThread) [custom_components.gree.climate] Setting SwhSlp: 0
2024-05-25 16:51:56.157 INFO (MainThread) [custom_components.gree.climate] Setting Lig: 1
2024-05-25 16:51:56.157 INFO (MainThread) [custom_components.gree.climate] Setting SwingLfRig: 0
2024-05-25 16:51:56.157 INFO (MainThread) [custom_components.gree.climate] Setting SwUpDn: 0
2024-05-25 16:51:56.157 INFO (MainThread) [custom_components.gree.climate] Setting Quiet: 0
2024-05-25 16:51:56.157 INFO (MainThread) [custom_components.gree.climate] Setting Tur: 0
2024-05-25 16:51:56.157 INFO (MainThread) [custom_components.gree.climate] Setting StHt: 0
2024-05-25 16:51:56.157 INFO (MainThread) [custom_components.gree.climate] Setting TemUn: 0
2024-05-25 16:51:56.157 INFO (MainThread) [custom_components.gree.climate] Setting HeatCoolType: 0
2024-05-25 16:51:56.158 INFO (MainThread) [custom_components.gree.climate] Setting TemRec: 0
2024-05-25 16:51:56.158 INFO (MainThread) [custom_components.gree.climate] Setting SvSt: 0
2024-05-25 16:51:56.158 INFO (MainThread) [custom_components.gree.climate] Setting SlpMod: 0
2024-05-25 16:51:56.158 INFO (MainThread) [custom_components.gree.climate] Done setting acOptions
2024-05-25 16:51:56.158 INFO (MainThread) [custom_components.gree.climate] HA target temp set according to HVAC state to: 25
2024-05-25 16:51:56.158 INFO (MainThread) [custom_components.gree.climate] HA lights option set according to HVAC state to: on
2024-05-25 16:51:56.158 INFO (MainThread) [custom_components.gree.climate] HA xfan option set according to HVAC state to: off
2024-05-25 16:51:56.158 INFO (MainThread) [custom_components.gree.climate] HA health option set according to HVAC state to: on
2024-05-25 16:51:56.158 INFO (MainThread) [custom_components.gree.climate] HA powersave option set according to HVAC state to: off
2024-05-25 16:51:56.158 INFO (MainThread) [custom_components.gree.climate] HA sleep option set according to HVAC state to: off
2024-05-25 16:51:56.158 INFO (MainThread) [custom_components.gree.climate] HA 8℃ heat option set according to HVAC state to: off
2024-05-25 16:51:56.159 INFO (MainThread) [custom_components.gree.climate] HA air option set according to HVAC state to: off
2024-05-25 16:51:56.159 INFO (MainThread) [custom_components.gree.climate] HA operation mode set according to HVAC state to: auto
2024-05-25 16:51:56.159 INFO (MainThread) [custom_components.gree.climate] HA swing mode set according to HVAC state to: Default
2024-05-25 16:51:56.159 INFO (MainThread) [custom_components.gree.climate] HA preset mode set according to HVAC state to: Default
2024-05-25 16:51:56.159 INFO (MainThread) [custom_components.gree.climate] HA fan mode set according to HVAC state to: Auto
2024-05-25 16:51:56.159 INFO (MainThread) [custom_components.gree.climate] Finished SyncState
2024-05-25 16:51:56.159 INFO (MainThread) [custom_components.gree.climate] supported_features(): 441
2024-05-25 16:51:56.159 INFO (MainThread) [custom_components.gree.climate] temperature_unit(): °C
2024-05-25 16:51:56.160 INFO (MainThread) [custom_components.gree.climate] hvac_modes(): [<HVACMode.AUTO: 'auto'>, <HVACMode.COOL: 'cool'>, <HVACMode.DRY: 'dry'>, <HVACMode.FAN_ONLY: 'fan_only'>, <HVACMode.HEAT: 'heat'>, <HVACMode.OFF: 'off'>]
2024-05-25 16:51:56.160 INFO (MainThread) [custom_components.gree.climate] min_temp(): 16
2024-05-25 16:51:56.160 INFO (MainThread) [custom_components.gree.climate] max_temp(): 30
2024-05-25 16:51:56.160 INFO (MainThread) [custom_components.gree.climate] target_temperature_step(): 1.0
2024-05-25 16:51:56.160 INFO (MainThread) [custom_components.gree.climate] fan_list(): ['Auto', 'Low', 'Medium-Low', 'Medium', 'Medium-High', 'High', 'Turbo', 'Quiet']
2024-05-25 16:51:56.160 INFO (MainThread) [custom_components.gree.climate] preset_modes(): ['Default', 'Full swing', 'Fixed in the leftmost position', 'Fixed in the middle-left position', 'Fixed in the middle postion', 'Fixed in the middle-right position', 'Fixed in the rightmost position']
2024-05-25 16:51:56.160 INFO (MainThread) [custom_components.gree.climate] swing_modes(): ['Default', 'Swing in full range', 'Fixed in the upmost position', 'Fixed in the middle-up position', 'Fixed in the middle position', 'Fixed in the middle-low position', 'Fixed in the lowest position', 'Swing in the downmost region', 'Swing in the middle-low region', 'Swing in the middle region', 'Swing in the middle-up region', 'Swing in the upmost region']
2024-05-25 16:51:56.160 INFO (MainThread) [custom_components.gree.climate] hvac_mode(): auto
2024-05-25 16:51:56.161 INFO (MainThread) [custom_components.gree.climate] supported_features(): 441
2024-05-25 16:51:56.161 INFO (MainThread) [custom_components.gree.climate] temperature_unit(): °C
2024-05-25 16:51:56.161 INFO (MainThread) [custom_components.gree.climate] current_temperature(): None
2024-05-25 16:51:56.161 INFO (MainThread) [custom_components.gree.climate] target_temperature(): 25
2024-05-25 16:51:56.161 INFO (MainThread) [custom_components.gree.climate] fan_mode(): Auto
2024-05-25 16:51:56.161 INFO (MainThread) [custom_components.gree.climate] preset_mode(): Default
2024-05-25 16:51:56.161 INFO (MainThread) [custom_components.gree.climate] swing_mode(): Default
2024-05-25 16:51:56.161 INFO (MainThread) [custom_components.gree.climate] name(): Small Room AC WIFI
2024-05-25 16:51:56.162 INFO (MainThread) [custom_components.gree.climate] supported_features(): 441
```

**Comment by toughvj** [comment]
This happened because the component was unable to obtain the key within 60 seconds. Probably the device was not available on the network. 

This is a bug in the code because retrieving the key should not happen in the class constructor. If the object does not retrieve the key when the class is constructed, the device becomes unusable until the next time the entire system is started.

**Comment by MukhaBoyka** [comment]
> This happened because the component was unable to obtain the key within 60 seconds. Probably the device was not available on the network.
> 
> This is a bug in the code because retrieving the key should not happen in the class constructor. If the object does not retrieve the key when the class is constructed, the device becomes unusable until the next time the entire system is started.

so what can i do? 

all three AC's are in gree app and work great from there.

**Comment by toughvj** [comment]
I'm working on it.

**Comment by toughvj** [comment]
solved with PR #179, can be closed
